src/extension.ts

```diff
@@ -21,10 +21,12 @@ class Extension {
     const debugEventManager = new DebugEventManager();
     debugEventManager.registerDebuggerPanel(debuggerPanel);
 
-    this.context.subscriptions.push(commands.registerCommand('visual-oo-debugger.openDebugger', () => debuggerPanel.openPanel()));
-    this.context.subscriptions.push(commands.registerCommand('visual-oo-debugger.exportPNG', () => debuggerPanel.exportPanel()));
-    this.context.subscriptions.push(commands.registerCommand('visual-oo-debugger.startGIF', () => debuggerPanel.startRecordingPanel()));
-    this.context.subscriptions.push(commands.registerCommand('visual-oo-debugger.stopGIF', () => debuggerPanel.stopRecordingPanel()));
+    this.registerCommand('visual-oo-debugger.openDebugger', () => debuggerPanel.openPanel());
+    this.registerCommand('visual-oo-debugger.exportPNG', () => debuggerPanel.exportPanel());
+    this.registerCommand('visual-oo-debugger.startGIF', () => debuggerPanel.startRecordingPanel());
+    this.registerCommand('visual-oo-debugger.stopGIF', () => debuggerPanel.stopRecordingPanel());
+    this.registerCommand('visual-oo-debugger.exportAsPlantUml', () => debuggerPanel.exportAsPlantUml());
+    this.registerCommand('visual-oo-debugger.exportAsGraphViz', () => debuggerPanel.exportAsGraphViz());
```

Comment:
These exports don't depend on the visualization, only on the data used. Is it really necessary to send this command to the debuggerPanel? Can't you just add the download logic here?

Reply:
Using download links has a few advantages:

* Simplicity
* New save dialogs are opened where the last file was saved

Using download links has a few disadvantages:

* Breaks the separation of concerns (poor architecture)
* No way to filter by file name extension(s)
* Escaping issues (for some programs the exported texts are simply cut off)

Reply:
Done: Using the save dialog API resolves all these issues

Reply:
DON'T involve the debuggerPanel here. The export has nothing to do with the panel. Now you have to implement `exportAsPlantUml` and `exportAsGraphViz` for every view.

Reply:
Or at the very latest, implement the export logic in the `debuggerPanel.ts`. But don't send the command to the `PanelViewProxy`.

Reply:
Good point. I will move the export logic to the `DebuggerPanel`, as I rely on the latest `PanelViewInput`.

Reply:
Done.